### ISSUE TITLE
vars not passed via `command` in workflows

### ISSUE THREAD

**Issue by jrnxf** [garden-io/garden#3286]
### Current Behavior

task:

```
- name: argtest
  command: [/bin/sh, -c]
  args: ["echo ${var.args || '-h'}"]
  ...
```

workflow:

```
kind: Workflow
name: workflow-a
description: workflow-a
steps:
  - description: WORKS
    script: garden run task argtest --var args="foobar"]
  - description: DOES NOT WORK
    command: [run, task, argtest, --var, args="foobar"]

```

### Expected behavior

I would think these would be exactly the same thing. Even stranger, this works!

`garden custom-workflow-b --var args="whyyyyy"`

```
kind: Command
name: custom-workflow-b
description:
  short: custom-workflow-b
gardenCommand:
  - run
  - task
  - argtest
  - --var
  - args="${var.args || ''}"
```

### Reproducible example
refer to above 

### Workaround

I guess I could call a custom workflow within a normal workflow? Idk, feels icky

### Additional context

Originally asked here: https://community.garden.io/t/why-does-one-of-these-work-while-the-other-doesnt/323

### Your environment

* **OS:** 
MacOS Monterey 12.6 (Mac M1 Pro 2021)
* **How I'm running Kubernetes:**  
locally via docker desktop

`garden version`
0.12.44

**Comment by thatvegandev** [comment]
@vvagaytsev I'd also like to point out that this isn't just for tasks. It's also for workflows. However there's an extra point of confusion because you actually can pass vars via command but NOTE when prefixed with `run, workflow, ...`.

```
  - description: Run refresh workflow
    command: [run, workflow, refresh, --var, arg="doesn't work"]
    command: [refresh, --var, arg="works"]
```

**Comment by salotz** [comment]
It is noted that you can't pass any "global" command line arguments to sub-commands in a workflow. I will +1 on supporting this explicitly as it makes a lot of sense to pass outputs of one step to the next. For instance running `build`, `test`, `publish`.

I didn't think of running it as a script though! That would have made things easier I suppose...

**Comment by 10ko** [comment]
This should be fixed in `0.13.x`, please re-open if the error is still present.
